DEVINFO.md

```diff
@@ -0,0 +1,174 @@
+
+Reviewers should aim to **perform a reviewer in one sitting** whenever possible.  This enables a reviewer to time-box their review, and ensures that by the time they finish studying the patch, they have a complete understanding of what the PR does in their head.  This, in turn, sets them up for success when writing up the acceptance plan.  It also enables reviewers to mark time for it on their calendars, which helps everyone else develop reasonable expectations as to when things will be done.
+
+Code reviews should be timely.  A PR review should begin no more than **2 business days** after the PR is submitted.  The `develop` and `next` branches in particular often change quickly, so letting a PR languish only creates more merge work for the submitter.  If a review cannot be begun within 2 business days, then the reviewers should **tell the submitter when they can begin**.  This gives the reviewer the opportunity to keep working on the PR (if needed) or even withdraw and resubmit it.
+
+Reviewers must, above all else, **ensure that submitters follow the PR checklist** below. 
+
+## Submitter Expectations
+
+Everyone is busy all the time with a host of different tasks.  Consequently, a PR's size and scope should be constrained so that **a review can be written for it no more than 2 hours.**  This time block starts when the reviewer opens the patch, and ends when the reviewer hits the "submit review" button.  If it takes more than 2 hours, then the PR should be broken into multiple PRs unless the reviewers agree to spend more time on it.  A PR can be rejected if the reviewers believe they will need longer than this.
+
+The size and scale of a PR depend on the reviewers' abilities to process the change.  Different reviewers and submitters have different levels of familiarity with the codebase.  Moreover, everyone has a different schedule -- sometimes, some people are more busy than others.  A successful PR submitter **takes the reviewers' familiarity and availability into account** when crafting the PR, even going to far as to ask in advance if a particular person could be available for review.
+
+## Submission Checklist
+
+A PR submission's text should **answer the following questions** for reviewers:
+
+* What problem is being solved by this PR?
+* What does the solution do to address them?
+* Why is this the best solution?  What alternatives were considered, and why are they worse?
+* What do reviewers need to be familiar with in order to provide useful feedback?
+* What issue(s) are addressed by this PR?
+* What are some hints to understanding some of the more intricate or clever parts of the PR?
```

Comment:
* Does this PR change an database schemas? Does a node need to re-sync from genesis to use this PR?

Reply:
Great, let's add that